envbuilder.go

```diff
@@ -99,6 +99,9 @@ func Run(ctx context.Context, options Options) error {
 	if options.InitCommand == "" {
 		options.InitCommand = "/bin/sh"
 	}
+	if options.CacheRepo == "" && options.PushImage {
+		return fmt.Errorf("--cache-repo must be set when using --push-image!")
+	}
```

Comment:
review: I don't see any good reason to not do this. It prevents potential head-scratching if someone sets `ENVBUILDER_PUSH_IMAGE` without specifying `ENVBUILDER_CACHE_REPO`.